### ISSUE TITLE
Support for Texture2DArray in opengl >= 3.0

### ISSUE THREAD

**Issue by PodeCaradox** [MonoGame/MonoGame#7348]
I just tested if the content pipline give me any errors with a shader that implement a Texture2DArray.
I got no errors, but i cant create a Texture2D with the array overload because of the implementation in the GraphicsCapabilities.OpenGL.cs class  
```
            // TODO: Implement OpenGL support for texture arrays
            // once we can author shaders that use texture arrays.
            SupportsTextureArrays = false;
```
Can this be changed or did i miss something?

#### What version of MonoGame does the bug occur on:
- MonoGame 3.8

#### What operating system are you using:
- Windows

#### What MonoGame platform are you using:
- DesktopGL

**Comment by minimalism** [comment]
I'm interested in this as well. When that comment was written it seems MojoShader could not compile texturearrays to HLSL: https://github.com/MonoGame/MonoGame/pull/3654
If that's no longer the case then I can maybe take a look at implementing the C# part.

**Comment by mrhelmut** [comment]
MojoShader is still around limiting things, but there are relevant progress here #7345, and some more related to shaders here #7352.

**Comment by minimalism** [comment]
Looks like all the relevant OpenGL calls needed to make this work are already known by MonoGame, you just need to wire them up in Texture2D.OpenGL.cs to create texture arrays. However building a pixel shader that samples a Texture2DArray gives the error "Unsupported texture type for ps_3_0" and raising the shader model to 4_0 gives  "Shader must be SM 3.0 or lower!"
I'll see if I can move forward with the ShaderConductor branch.

**Comment by mrhelmut** [comment]
Yeah, that's because of MojoShader, it presently is the limiting component in OpenGL shader processing. It is likely a safe bet to build upon @cpt-max's work.

**Comment by minimalism** [comment]
Yep that works. I'll tidy it up and create a PR but if anyone needs texture arrays urgently, they are here https://github.com/MonoGame/MonoGame/commit/fbf878a176d0a76fbfa25eb2a9bac72f57748182
